### PR TITLE
CORE-8066: add the system ID to calls to converted endpoints.

### PR DESCRIPTION
**Pull request by dennisroberts71**

### CODE REVIEW

File: de-lib/src/main/java/org/iplantc/de/client/models/apps/App.java [hunk 1]
```diff
@@ -1,23 +1,20 @@
 package org.iplantc.de.client.models.apps;
 
+import com.google.gwt.user.client.ui.HasName;
```
Comment (ash3rz) [inline de-lib/src/main/java/org/iplantc/de/client/models/apps/App.java]: Totally trivial, but we have some [code style settings](https://github.com/cyverse-de/ui/tree/master/iplant/code-style) that make sure the iplant/cyverse imports come before anything else.

File: de-lib/src/main/java/org/iplantc/de/desktop/client/presenter/DesktopPresenterWindowEventHandler.java [hunk 2]
```diff
@@ -219,7 +221,7 @@ public void onRequestUploadFromUrl(RequestImportFromUrlEvent event) {
 
     @Override
     public void onRunAppActionInitiated(RunAppEvent event) {
-        AppWizardConfig config = ConfigFactory.appWizardConfig(event.getAppToRun().getId());
+        AppWizardConfig config = ConfigFactory.appWizardConfig(event.getAppToRun());
```
Comment (ash3rz) [inline de-lib/src/main/java/org/iplantc/de/desktop/client/presenter/DesktopPresenterWindowEventHandler.java]: Was the .getId() supposed to be dropped from here?

Comment (dennisroberts71) [inline de-lib/src/main/java/org/iplantc/de/desktop/client/presenter/DesktopPresenterWindowEventHandler.java]: It was. I added an overloaded version of `ConfigFactory.appWizardConfig` that takes an implementation of `HasQualifiedId`, which is implemented by `App`.

File: de-lib/src/main/java/org/iplantc/de/client/models/apps/App.java (hunk 1 above)
Comment (dennisroberts71) [inline de-lib/src/main/java/org/iplantc/de/client/models/apps/App.java]: Cool, I'll reformat the code shortly.

File: de-lib/src/main/java/org/iplantc/de/desktop/client/presenter/DesktopPresenterWindowEventHandler.java (hunk 2 above)
Comment (ash3rz) [inline de-lib/src/main/java/org/iplantc/de/desktop/client/presenter/DesktopPresenterWindowEventHandler.java]: Ah, cool, I missed that. Thanks!

**Comment by dennisroberts71** [comment]
Thanks for the review! I'll wait until the reviews are done for the corresponding service changes before I merge this.

**Comment by dennisroberts71** [comment]
I'm closing this PR so that I can open a new one against master later this week.

**Pull request closed without merging**